blocks/v2-explore-articles/v2-explore-articles.js

```diff
@@ -109,67 +135,369 @@ const buildArticlesTemplate = (articles) =>
     <div class="${CLASSES.collage}">
       ${buildArticlesTemplate(articles)}
     </div>
   </div>
   <div class="${CLASSES.showMoreButtonWrapper}">
-    <button class="${CLASSES.showMoreButton} button button--secondary button--large">
-      ${LABELS.SHOW_MORE}
-    </button>
+    ${buildShowMoreBtn()}
+  </div>
+  <div class="${CLASSES.noArticlesMessage} hide">
+    <h3>${LABELS.ERROR_TITLE}</h3>
+    <p>${LABELS.ERROR_TEXT}</p>
   </div>
 `);
 
+  return template;
+};
+
+// Updates a single HTML element
+const updateHtmlElmt = (block, selectedClass, newEl) => {
+  const container = block.querySelector(`.${selectedClass}`);
+  const newFragment = docRange.createContextualFragment(newEl);
+  container.innerHTML = '';
+  container.append(newFragment);
+};
+
+// Update the article list with the global object "appliedFilters"
+const updateArticleList = async (block, offset = 0) => {
+  const { articles: newFilteredArticles, count } = await getData(appliedFilters, offset);
+  totalArticleCount = count;
+
+  const newExtraLine = buildFiltersExtraLine(defaultAmountOfArticles, totalArticleCount);
+  updateHtmlElmt(block, CLASSES.extraLine, newExtraLine);
+
+  const newArticlesTemplate = buildArticlesTemplate(newFilteredArticles);
+  updateHtmlElmt(block, CLASSES.collage, newArticlesTemplate);
+
+  const showMoreBtnWrapper = block.querySelector(`.${CLASSES.showMoreButtonWrapper}`);
+  showMoreBtnWrapper.classList[newFilteredArticles.length >= count && count < defaultAmountOfArticles ? 'add' : 'remove']('hide');
+
+  const noArticlesMessage = block.querySelector(`.${CLASSES.noArticlesMessage}`);
+  noArticlesMessage.classList[count == 0 ? 'remove' : 'add']('hide');
+};
+
+// Check toggle buttons to show or hide applied filters
+const handleToggleBtns = (list, filters) => {
+  if (isMobile) {
+    return;
+  }
+  const toggleMoreBtn = filters.parentElement.querySelector(`.${CLASSES.toggleMore}`);
+  const toggleLessBtn = filters.parentElement.querySelector(`.${CLASSES.toggleLess}`);
+
+  const toggleMoreBtnWidth = toggleMoreBtn.getBoundingClientRect().width;
+  const { width: listWidth, height: listHeight } = filters.getBoundingClientRect();
+  const wrapperWidth = list.parentElement.getBoundingClientRect().width;
+
+  const maxListWidth = wrapperWidth * 0.75 - toggleMoreBtnWidth; // set the available space in the wrapper
+
+  const listBiggerThanWrapper = listWidth >= maxListWidth; // if list is wider than the container
+  const isMultiLine = listHeight > 36; // this is the height of the filter wrapper's height in CSS
+
+  // Logic that shows/hides the filter toggle buttons
+  if (listBiggerThanWrapper) {
+    if (isMultiLine) {
+      toggleMoreBtn.classList.add('hide');
+      toggleLessBtn.classList.remove('hide');
+    } else {
+      toggleMoreBtn.classList.remove('hide');
+      toggleLessBtn.classList.add('hide');
+    }
+  } else {
+    toggleMoreBtn.classList.add('hide');
+    toggleLessBtn.classList.add('hide');
+  }
```

Comment:
this conditional can be resumed as the following:

```suggestion
toggleMoreBtn.classList.toggle('hide', !listBiggerThanWrapper || isMultiLine);
toggleLessBtn.classList.toggle('hide', !listBiggerThanWrapper || !isMultiLine);
```

Reply:
the toggle method can add conditionaly a class

Reply:
makes sense.. but for some reason does not work 🤷